clients/dataset/dataset.go

```diff
@@ -122,6 +122,54 @@ func (c *Client) Get(ctx context.Context, id string) (m Model, err error) {
 	return
 }
 
+// GetByPath returns dataset level information for a given dataset path
+func (c *Client) GetByPath(ctx context.Context, path string) (m Model, err error) {
+	uri := fmt.Sprintf("%s/%s", c.url, path)
```

Comment:
might be worth sanitising the provided path and checking for preceding '/' if we're going to insert one here